src/common/comp/brute_force.cc

```diff
@@ -152,22 +141,16 @@ BruteForce::SearchWithBuf(const DataSetPtr base_dataset, const DataSetPtr query_
     auto nq = query_dataset->GetRows();
 
     BruteForceConfig cfg;
-    auto load_res = Config::Load(cfg, config, knowhere::SEARCH);
-    if (load_res != Status::success) {
-        return load_res;
-    }
-
-    auto metric_type = Str2FaissMetricType(cfg.metric_type);
-    if (!metric_type.has_value()) {
-        LOG_KNOWHERE_ERROR_ << "Invalid metric type: " << cfg.metric_type;
-        return Status::invalid_metric_type;
-    }
+    RETURN_IF_ERROR(Config::Load(cfg, config, knowhere::SEARCH));
 
     int topk = cfg.k;
     auto labels = ids;
     auto distances = dis;
 
-    auto faiss_metric_type = metric_type.value();
+    ASSIGN_OR_RETURN(faiss::MetricType, faiss_metric_type,
```

Comment:
if there is LOG, suggest keep the code as it be

Reply:
Done, removed logging from ASSIGN_OR_RETURN